nifi-nar-bundles/nifi-apicurio-bundle/nifi-apicurio-schema-registry-service/src/main/java/org/apache/nifi/apicurio/schemaregistry/ApicurioSchemaRegistry.java

```diff
@@ -0,0 +1,135 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.apicurio.schemaregistry;
+
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.annotation.lifecycle.OnEnabled;
+import org.apache.nifi.apicurio.schemaregistry.client.ApicurioSchemaRegistryClient;
+import org.apache.nifi.apicurio.schemaregistry.client.CachingSchemaRegistryClient;
+import org.apache.nifi.apicurio.schemaregistry.client.SchemaRegistryApiClient;
+import org.apache.nifi.apicurio.schemaregistry.client.SchemaRegistryClient;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.controller.AbstractControllerService;
+import org.apache.nifi.controller.ConfigurationContext;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.schema.access.SchemaField;
+import org.apache.nifi.schema.access.SchemaNotFoundException;
+import org.apache.nifi.schemaregistry.services.SchemaRegistry;
+import org.apache.nifi.serialization.record.RecordSchema;
+import org.apache.nifi.serialization.record.SchemaIdentifier;
+import org.apache.nifi.web.client.provider.api.WebClientServiceProvider;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.OptionalInt;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+@Tags({"schema", "registry", "apicurio", "avro"})
+@CapabilityDescription("Provides a Schema Registry that interacts with the Apicurio Schema Registry so that those Schemas that are stored in the Apicurio Schema "
+        + "Registry can be used in NiFi. When a Schema is looked up by name by this registry, it will find a Schema in the Apicurio Schema Registry with their names.")
+public class ApicurioSchemaRegistry extends AbstractControllerService implements SchemaRegistry {
+
+    private static final Set<SchemaField> schemaFields = EnumSet.of(SchemaField.SCHEMA_NAME, SchemaField.SCHEMA_TEXT,
+            SchemaField.SCHEMA_TEXT_FORMAT, SchemaField.SCHEMA_IDENTIFIER, SchemaField.SCHEMA_VERSION);
+
+    static final PropertyDescriptor SCHEMA_REGISTRY_URL = new PropertyDescriptor.Builder()
+            .name("schema-registry-url")
+            .displayName("Schema Registry URL")
+            .description("The URL of the Schema Registry e.g. http://localhost:8080")
+            .addValidator(StandardValidators.URL_VALIDATOR)
+            .required(true)
+            .build();
+    static final PropertyDescriptor CACHE_SIZE = new PropertyDescriptor.Builder()
+            .name("cache-size")
+            .displayName("Cache Size")
+            .description("Specifies how many Schemas should be cached from the Schema Registry")
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
```

Comment:
AFAIK `NON_NEGATIVE_INTEGER_VALIDATOR` validates zero (0) value. Is that really a valid value for 'Cache Size'? If yes, it might worth it to mention its meaning in the description.